From e660cee0e3f94dabce5ecf8ae7a05a342b8ed464 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 28 Nov 2022 22:38:58 +0000
Subject: [PATCH 2/2] gnu: python-sgp4: Move to astronomy.

* gnu/packages/astronomy.scm: (python-sgp4): Move here from python-science.scm.
  [description]: Expand description, annotate acronyms, add link to CelesTrak.
---
 gnu/packages/astronomy.scm      | 29 +++++++++++++++++++++++++++++
 gnu/packages/python-science.scm | 20 --------------------
 2 files changed, 29 insertions(+), 20 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 439fdf72a8..0fb3efff0a 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -1412,6 +1412,35 @@ (define-public python-reproject
 changing the pixel resolution, orientation, coordinate system.")
     (license license:bsd-3)))
 
+(define-public python-sgp4
+  (package
+    (name "python-sgp4")
+    (version "2.21")
+    (source
+     (origin
+       (method url-fetch)
+       (uri (pypi-uri "sgp4" version))
+       (sha256
+        (base32 "1vzcrlrlzmhbycdz16m8v241l8zx49vsy81wcd0yjxs80isvhyb1"))))
+    (build-system python-build-system)
+    (propagated-inputs
+     (list python-numpy))
+    (home-page "https://github.com/brandon-rhodes/python-sgp4")
+    (synopsis "Track earth satellite TLE orbits using SGP4")
+    (description
+     "This package provides a Python implementation for computations of the
+position and velocity of an earth-orbiting satellite, given the satellite’s
+@acronym{TLE, Two-line element set} orbital elements from a source like
+@url{CelesTrak, https://celestrak.org/}.
+
+It implements the most recent version of @acronym{SGP4, Simplified General
+Perturbation models}, and is regularly run against the SGP4 test suite to make
+sure that its satellite position predictions agree to within 0.1 mm with the
+predictions of the standard distribution of the algorithm.  This error is far
+less than the 1–3 km/day by which satellites themselves deviate from the ideal
+orbits described in TLE files.")
+    (license license:expat)))
+
 (define-public python-sunpy
   (package
     (name "python-sunpy")
diff --git a/gnu/packages/python-science.scm b/gnu/packages/python-science.scm
index 7af3f8202a..4e9582191c 100644
--- a/gnu/packages/python-science.scm
+++ b/gnu/packages/python-science.scm
@@ -290,26 +290,6 @@ (define-public python-scikit-allel
 genetic variation data.")
     (license license:expat)))
 
-(define-public python-sgp4
-  (package
-    (name "python-sgp4")
-    (version "2.21")
-    (source
-     (origin
-       (method url-fetch)
-       (uri (pypi-uri "sgp4" version))
-       (sha256
-        (base32 "1vzcrlrlzmhbycdz16m8v241l8zx49vsy81wcd0yjxs80isvhyb1"))))
-    (build-system python-build-system)
-    (propagated-inputs
-     (list python-numpy))
-    (home-page "https://github.com/brandon-rhodes/python-sgp4")
-    (synopsis "Track earth satellite TLE orbits using SGP4")
-    (description
-     "This package provides a Python implementation of the most recent version
-of the SGP4 satellite tracking algorithm.")
-    (license license:expat)))
-
 (define-public python-trimesh
   (package
     (name "python-trimesh")
-- 
2.38.1

